controllers/nstemplateset/cluster_resources.go

```diff
@@ -103,6 +103,7 @@ func (r *clusterResourcesManager) apply(ctx context.Context, nsTmplSet *toolchai
 		toolchainv1alpha1.TemplateRefLabelKey: tierTemplate.templateRef,
 		toolchainv1alpha1.TierLabelKey:        tierTemplate.tierName,
 		toolchainv1alpha1.ProviderLabelKey:    toolchainv1alpha1.ProviderLabelValue,
+		toolchainv1alpha1.OwnerLabelKey:       nsTmplSet.Namespace,
```

Comment:
It took me a good while to realize that `nsTmplSet.Namespace` is used here to distinguish the member operators in the cluster. Maybe this could be worth a comment.

But more importantly, I'm not sure this correct.

The assumption that we make about the cluster resources managed by NSTemplateSet is that they are **specific** to the Space for which they have been deployed (which implies the `${SPACE_NAME}` in their `metadata.name` btw.) and so this kind of conflict should never actually happen. If it does, then I think the bug is actually in the tests, not in this code.

The assumption is mentioned here: https://github.com/codeready-toolchain/member-operator/blob/master/controllers/nstemplateset/cluster_resources.go#L25